src/mon/MonCommands.h

```diff
@@ -447,7 +447,8 @@ COMMAND("osd dump " \
 	"print summary of OSD map", "osd", "r", "cli,rest")
 COMMAND("osd tree " \
 	"name=epoch,type=CephInt,range=0,req=false " \
-	"name=states,type=CephChoices,strings=up|down|in|out,n=N,req=false", \
+        "name=states,type=CephChoices,strings=up|down|in|out,n=N,req=false " \
+        "name=shadow,type=CephChoices,strings=--show-shadow,req=false", \
```

Comment:
Can you add a qa/workunits/cephtool/test.sh case that verifies this works?  (includes ~class with flag, doesn't include it without?)  thanks!